Logical operators

vahmelk99 opened the pull request:
Please check my translation.

vahmelk99 committed:
article.md to line 19

vahmelk99 committed:
OR operator ended

vahmelk99 committed:
&& translated

vahmelk99 committed:
article added

vahmelk99 committed:
importance translated

vahmelk99 committed:
1-alert-null-2-undefined translated

vahmelk99 committed:
3-alert-1-null-2 translated

vahmelk99 committed:
2-alert-or translated

vahmelk99 committed:
2-alert-or translated

vahmelk99 committed:
4-alert-and translated

vahmelk99 committed:
5-alert-and-or translated

vahmelk99 committed:
titles rename and 6th ex.

vahmelk99 committed:
7-check-if-out-range translated

vahmelk99 committed:
8-if-question translated

vahmelk99 committed:
start translating 9-check-login

vahmelk99 committed:
Merge branch 'master' of https://github.com/vahmelk99/hy.javascript.info

vahmelk99 committed:
9-check-login translated

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/javascript-tutorial/hy.javascript.info?pullRequest=82) <br/>All committers have signed the CLA.

bugron commented:
Thanks for the PR @vahmelk99. I'll be reviewing it soon.

bugron commented on 1-js/02-first-steps/11-logical-operators/1-alert-null-2-undefined/solution.md:
````diff
@@ -1,6 +1,5 @@
 
 ```js run
-alert( null || 2 || undefined );
+alert(null || 2 || undefined)
````
Please restore the original formatting of this line.

bugron commented on 1-js/02-first-steps/11-logical-operators/1-alert-null-2-undefined/task.md:
```diff
@@ -1,12 +1,11 @@
-importance: 5
+Կարևորություն: 5
```
You should not translate lines similar to this one. Please, restore it.

bugron commented on 1-js/02-first-steps/11-logical-operators/1-alert-null-2-undefined/task.md:
````diff
@@ -1,12 +1,11 @@
 
 ```js
-alert( null || 2 || undefined );
+alert(null || 2 || undefined)
````
Please restore the original formatting of this line.

bugron commented on 1-js/02-first-steps/11-logical-operators/2-alert-or/task.md:
```diff
@@ -1,12 +1,11 @@
-importance: 3
+Կարևորություն: 3
```
^

bugron commented on 1-js/02-first-steps/11-logical-operators/2-alert-or/task.md:
````diff
@@ -1,12 +1,11 @@
 
 ```js
-alert( alert(1) || 2 || alert(3) );
+alert(alert(1) || 2 || alert(3))
````
^

bugron commented on 1-js/02-first-steps/11-logical-operators/2-alert-or/solution.md:
````diff
@@ -1,13 +1,13 @@
 
 ```js run
-alert( alert(1) || 2 || alert(3) );
+alert(alert(1) || 2 || alert(3))
````
^

bugron commented on 1-js/02-first-steps/11-logical-operators/2-alert-or/solution.md:
````diff
@@ -1,13 +1,13 @@
 ```
 
-The call to `alert` does not return a value. Or, in other words, it returns `undefined`.
+`alert`֊ի կանչը չի վերադարձնում ոչ մի արժեք։ Կամ այլ կերպա ասած վերադարձնում է `undefined`։
````
You have a typo in this sentence.

bugron commented on 1-js/02-first-steps/11-logical-operators/2-alert-or/solution.md:
```diff
@@ -1,13 +1,13 @@
-3. The second operand `2` is truthy, so the execution is halted, `2` is returned and then shown by the outer alert.
+1. Սկզբում ԿԱՄ `||`֊ը հաշվում(evaluates) է իր ձախ կողմի օպերանդը՝ `alert(1)`, որն էլ էկրանին ցույց է տալիս `1` արժեքը։
+2. `alert`֊ը վերադարձնում է `undefined`։ ԿԱՄ֊ը անցնում է երկրորդ օպերանդին՝ որոնելով ճշմարիտ արժեք։
+3. Երկրորդ օպերանդը `2` է, որը ճշմարիտ է, և ԿԱՄ֊ը դադարացնում է իր հաշվարկները։ `2`֊ը վերադարձվում է և ցուցադրվում էկրանին դրսի `alert`֊ի միջոցով։
```
You have a typo in this sentence.

bugron commented on 1-js/02-first-steps/11-logical-operators/3-alert-1-null-2/solution.md:
````diff
@@ -1,6 +1,6 @@
 
 ```js run
-alert( 1 && null && 2 );
+alert(1 && null && 2)
````
^

bugron commented on 1-js/02-first-steps/11-logical-operators/3-alert-1-null-2/task.md:
```diff
@@ -1,12 +1,11 @@
-importance: 5
+Կարևորություն: 5
```
^

bugron commented on 1-js/02-first-steps/11-logical-operators/3-alert-1-null-2/task.md:
````diff
@@ -1,12 +1,11 @@
 
 ```js
-alert( 1 && null && 2 );
+alert(1 && null && 2)
````
^

bugron commented on 1-js/02-first-steps/11-logical-operators/4-alert-and/solution.md:
````diff
@@ -1,10 +1,10 @@
 
 ```js run
-alert( alert(1) && alert(2) );
+alert(alert(1) && alert(2));
````
^

bugron commented on 1-js/02-first-steps/11-logical-operators/4-alert-and/task.md:
```diff
@@ -1,12 +1,11 @@
-importance: 3
+Կարևորություն: 3
```
^

bugron commented on 1-js/02-first-steps/11-logical-operators/4-alert-and/task.md:
````diff
@@ -1,12 +1,11 @@
 
 ```js
-alert( alert(1) && alert(2) );
+alert(alert(1) && alert(2))
````
^

bugron commented on 1-js/02-first-steps/11-logical-operators/5-alert-and-or/task.md:
```diff
@@ -1,12 +1,11 @@
-importance: 5
+Կարևորություն: 5
```
^

bugron commented on 1-js/02-first-steps/11-logical-operators/6-check-if-in-range/task.md:
```diff
@@ -1,9 +1,9 @@
-importance: 3
+Կարևորություն: 3
```
^

bugron commented on 1-js/02-first-steps/11-logical-operators/7-check-if-out-range/task.md:
```diff
@@ -1,9 +1,9 @@
-importance: 3
+Կարևորություն: 3
```
^

bugron commented on 1-js/02-first-steps/11-logical-operators/8-if-question/task.md:
```diff
@@ -1,16 +1,15 @@
-importance: 5
+Կարևորություն: 5
```
^

bugron commented on 1-js/02-first-steps/11-logical-operators/8-if-question/task.md:
````diff
@@ -1,16 +1,15 @@
+if (-1 && 0) alert('երկրորդ')
+if (null || -1 && 1) alert('երրորդ')
 ```
-
````
Please restore the new line at the end of this file.

bugron replied:
Same thing here.

bugron commented on 1-js/02-first-steps/11-logical-operators/5-alert-and-or/task.md:
````diff
@@ -1,12 +1,11 @@
 ```js
 alert( null || 2 && 3 || 4 );
 ```
-
````
Please restore the new line at the end of this file.

bugron replied:
Also here.

bugron commented on 1-js/02-first-steps/11-logical-operators/4-alert-and/task.md:
````diff
@@ -1,12 +1,11 @@
-alert( alert(1) && alert(2) );
+alert(alert(1) && alert(2))
 ```
-
````
Please restore the new line at the end of this file.

bugron replied:
Also here. It appears you editor might mess up with new lines. Please restore.

bugron commented on 1-js/02-first-steps/11-logical-operators/3-alert-1-null-2/task.md:
````diff
@@ -1,12 +1,11 @@
-alert( 1 && null && 2 );
+alert(1 && null && 2)
 ```
-
````
Please restore the new line at the end of this line.

bugron replied:
Same thing here.

bugron commented on 1-js/02-first-steps/11-logical-operators/2-alert-or/task.md:
````diff
@@ -1,12 +1,11 @@
-alert( alert(1) || 2 || alert(3) );
+alert(alert(1) || 2 || alert(3))
 ```
-
````
Please restore the new line at the end of this file.

bugron replied:
Same thing here.

bugron commented on 1-js/02-first-steps/11-logical-operators/1-alert-null-2-undefined/task.md:
````diff
@@ -1,12 +1,11 @@
-alert( null || 2 || undefined );
+alert(null || 2 || undefined)
 ```
-
````
Please restore the new line at the end of this file.

bugron replied:
Same thing with the new line here.

bugron commented on 1-js/02-first-steps/11-logical-operators/1-alert-null-2-undefined/solution.md:
````diff
@@ -1,6 +1,5 @@
-alert( null || 2 || undefined );
+alert(null || 2 || undefined)
 ```
-
````
Please restore the new line at the end of this file.

bugron replied:
@vahmelk99 did you restore the new line at the end of this file because GitHub does not show it is?

vahmelk99 replied:
![image](https://user-images.githubusercontent.com/30202111/129405856-d4a60ed9-5eb8-432c-b7b4-f189fb83a56e.png)
![image](https://user-images.githubusercontent.com/30202111/129405965-66cf1303-dc0b-4973-94c5-f48b4af2539e.png)

@bugron I don't know why it isn't showing here :/ In my local repo file has 6 lines. And problem not only for this file. There are another files like this.

bugron commented on 1-js/02-first-steps/11-logical-operators/9-check-login/ifelse_task.svg:
```diff
@@ -1 +1,206 @@
-<svg xmlns="http://www.w3.org/2000/svg" width="599" height="493" viewBox="0 0 599 493"><defs><style>@import url(https://fonts.googleapis.com/css?family=Open+Sans:bold,italic,bolditalic%7CPT+Mono);@font-face{font-family:&apos;PT Mono&apos;;font-weight:700;font-style:normal;src:local(&apos;PT MonoBold&apos;),url(/font/PTMonoBold.woff2) format(&apos;woff2&apos;),url(/font/PTMonoBold.woff) format(&ap ...
```
Please restore the original SVG file.

bugron replied:
I don't see this file is restored. If there is something to translate in this SVG please take a look at the README.md file. You'll find some instructions on how to do that.

bugron commented on 1-js/02-first-steps/11-logical-operators/9-check-login/solution.md:
````diff
@@ -1,25 +1,25 @@
 ```
 
-Note the vertical indents inside the `if` blocks. They are technically not required, but make the code more readable.
+Ուշադրություն դարձրեք ուղղահայաց խորություններին(indents) `if` բլոկների ներսում։ Դրանք պարտադիր չեն, բայց դարձնում են ծրագիրը ավելի ընոեռնելի։
````
You have a couple of typos here.

bugron commented on 1-js/02-first-steps/11-logical-operators/9-check-login/task.md:
```diff
@@ -1,25 +1,25 @@
-importance: 3
+Կարևորություն: 3
```
^

bugron commented on 1-js/02-first-steps/11-logical-operators/9-check-login/task.md:
```diff
@@ -1,25 +1,25 @@
+# Մուտքի ստուգում (login)
 
-Write the code which asks for a login with `prompt`.
+Գրեք ծրագիր, որը կպահանջի մուտքի օգտագործելով `prompt`֊ը։
```
You have a typo here.

bugron commented on 1-js/02-first-steps/11-logical-operators/9-check-login/task.md:
```diff
@@ -1,25 +1,25 @@
-- For an empty string or cancelled input, show "Canceled"
+- Եթե այն "Գլխավոր" է, ապա ցուցադրվում է "Բարի գալուստ",
+- Ուրիշ տողային արժեքի դեպքում ցուցադրվում է "Սխալ գաղտնաբառ",
+- Դատարի տողի և չեղարկված(cancelled) դեպքերում ցուցադրվում է "Չեղարկված է"
```
Also a typo here.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
+# Տրամաբանական օպերատորներ (Logical operators)
 
-There are four logical operators in JavaScript: `||` (OR), `&&` (AND), `!` (NOT), `??` (Nullish Coalescing). Here we cover the first three, the `??` operator is in the next article.
+JavaScript-ում կան չորս տրամաբանական օպերատորներ՝ `||` (ԿԱՄ), `&&` (ԵՎ), `!` (ՈՉ), `??` (Null-ի հետ համակցման). Այստեղ մենք կծանոթանանք առաջին երեքին։ `??` օպերատորին կարող եք ծանոթանալ հաջորդ հոդվածում։
```
```suggestion
JavaScript-ում կան չորս տրամաբանական օպերատորներ՝ `||` (ԿԱՄ), `&&` (ԵՎ), `!` (ՈՉ), `??` (Nullish Coalescing). Այստեղ մենք կծանոթանանք առաջին երեքին։ `??` օպերատորին կարող եք ծանոթանալ հաջորդ հոդվածում։
```
I'd prefer to go with the original name of this operator or at least also supply the English original name.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
+JavaScript-ում կան չորս տրամաբանական օպերատորներ՝ `||` (ԿԱՄ), `&&` (ԵՎ), `!` (ՈՉ), `??` (Null-ի հետ համակցման). Այստեղ մենք կծանոթանանք առաջին երեքին։ `??` օպերատորին կարող եք ծանոթանալ հաջորդ հոդվածում։
 
-Although they are called "logical", they can be applied to values of any type, not only boolean. Their result can also be of any type.
+Չնայած նրանք անվանվում են "տրամաբանական", նրանք կարող են կիրառվել կամայական(տարբերվող տրամաբանական տիպից) տիպի վրա։ Նրանց արդյունքը նույպես կարող է լինել կամայական տիպի։
```
You have a typo here.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
````diff
@@ -1,289 +1,290 @@
 ```
 
-In classical programming, the logical OR is meant to manipulate boolean values only. If any of its arguments are `true`, it returns `true`, otherwise it returns `false`.
+Դասական ծրագրավորման մեջ տրամաբանական ԿԱՄ օպերատորը կիրառվում է միայն տրամաբանական արժեքների վրա։ Եթե նրա արգումենտներից մեկը ճշմարի `true` է, ապա այն վերադարձնում է `true`, հակառակ դեպքում՝ `false`։
````
You have a typo here.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
+alert(true || true) // true
+alert(false || true) // true
+alert(true || false) // true
+alert(false || false) // false
```
Please restore original formatting of those 4 lines.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
````diff
@@ -1,289 +1,290 @@
 ```
 
-As we can see, the result is always `true` except for the case when both operands are `false`.
+Ինչպես տեսնում ենք արդյունքը միշտ ճիշտ(`true`) է, բացի այն դեպքից, երբ երկու օպերանդներն էլ սխալ(`false`) են։
````
Please put spaces before parentheses. @vahmelk99 make sure to check other places too.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
-  alert( 'truthy!' );
+if (1 || 0) {
+  // աշխատում է ինչպես if( true || false )
+  alert('Ճշմարիտ է!')
```
Please put the line `// աշխատում է ինչպես if( true || false )` at its original place (the line before) to not introduce a line shift. See English text.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
+1. **Առաջին ճշմարիտ արժեքի ստացումը փոփոխականների կամ արտահայտությունների ցուցակից։**
 
-    For instance, we have `firstName`, `lastName` and `nickName` variables, all optional (i.e. can be undefined or have falsy values).
+   Օրինակի, ունենք `firstName`, `lastName` և `nickName` փոփոխականները, բոլորը ըստ ցանկության կարող են ստանալ սխալական կամ `undefined` արժեքներ։
```
You have a typo here.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
````diff
@@ -1,289 +1,290 @@
+   *!*
+   alert( firstName || lastName || nickName || "Anonymous"); // SuperCoder
+   */!*
+   ```
````
Please restore original indentation of these lines.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
````diff
@@ -1,289 +1,290 @@
+   ```js run no-beautify
+   *!*true*/!* || alert("չի երևա");
+   *!*false*/!* || alert("կերևա");
+   ```
````
Please restore the indentation of these lines.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
+alert(true && true) // true
+alert(false && true) // false
+alert(true && false) // false
+alert(false && false) // false
```
Restore original formatting of these lines.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
+2. **Կրճատ հաշվարկ (Short-circuit evaluation)։**
 
-    Another feature of OR `||` operator is the so-called "short-circuit" evaluation.
+   ԿԱՄ `||`֊ի մեկ այլ հատկությունը այսպես կոչված "կրճատ հաշվարկ"֊ի հատկությունն է։
```
I can see you removed some white space at the beginning of this and next sentences up to the section `## && (AND)`. Please restore them.

vahmelk99 replied:
That was my auto save formatter :D I'll turn it off in the next translations)

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
-if (1 && 0) { // evaluated as true && false
-  alert( "won't work, because the result is falsy" );
+if (1 && 0) {
+  // հաշվվում է ինչպես true && false
```
Please reset the original place of this line.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
+- Ամեն օպերանդի արժեք փոխակերպում է տրամաբանական տիպի։ Եթե արդյունքը `false` է այն կանգ է առնում և վերադարձնում այդ օպերանդի օրիգինալ արժեքը։
+- Եթե բոլոր օպերանդները հաշվվում են, և բոլորի արժեքը `true` է լինում, ապա վերադարձնում է վերջին օպերանդի արժեքը։
 
 In other words, AND returns the first falsy value or the last value if none were found.
```
This line is not translated.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
+
+// Եթե առաջին օպերանդը սխալական է,
+// ապա ԵՎ֊ը վերադարձնում է այն։
+// Երկրորդ օպերանդը անտեսվում է։
```
Please do not introduce unnecessary line breaks.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
``````diff
@@ -1,289 +1,290 @@
 
-So the code `a && b || c && d` is essentially the same as if the `&&` expressions were in parentheses: `(a && b) || (c && d)`.
-````
+`````
``````
You have an extra backtick here.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
````diff
@@ -1,289 +1,290 @@
 ```
 
-The action in the right part of `&&` would execute only if the evaluation reaches it. That is, only if `(x > 0)` is true.
+Աջ կողմի գործողությունը կկատարվի այն ժամանակ, երբ ձախից աջ արժեքավրումը հասնի իրեն(այսինքն ձախ կողմի բոլոր արժեքները ճշմարիտ են)։ Դա տեղի կունենա, եթե `(x > 0)` արտահայտությունը ճիշտ լինի։
````
You have a typo here.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
``````diff
@@ -1,289 +1,290 @@
-````
-
+Թեև `&&`֊ով տարբերակն ավելի կարճ է, `if`֊ը ավելի հասկանալի է և ավելի ընթեռնելի։ Այսպիսով, մենք խորհուրդ ենք տալիս օգտագործել յուրաքանչյուր գրելաձև իր նշանակությանը համապատասխան՝ օգտագործեք `if` եթե ցանկանում եք պայման ստուգել, և `&&`, եթե ցանկանում եք կատարել ԵՎ տրամաբանական գործողությունը։
+`````
``````
An extra backtick here also.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
````diff
@@ -1,289 +1,290 @@
 ```
 
-The operator accepts a single argument and does the following:
+Օպերատորը ընդունում է մեկ արգումենտ է կատարում հետևյալ քայլերը․
````
A typo here.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -1,289 +1,290 @@
-alert( !true ); // false
-alert( !0 ); // true
+alert(!true) // false
+alert(!0) // true
```
Restore original formatting.

bugron reviewed (changes requested):
Please do not be discouraged by the number of comments. It is normal for this project. Once the suggested changes are implemented I'll give it another round of reviews. Thanks.

bugron commented:
@vahmelk99 also for future PRs the name of the PR should exactly match the title of the article you are translating (see https://github.com/javascript-tutorial/hy.javascript.info/issues/1). For example the title of this PR should have been `Logical operators`.

vahmelk99 committed:
translation bugfixes in logical operators

vahmelk99 commented:
> @vahmelk99 also for future PRs the name of the PR should exactly match the title of the article you are translating (see #1). For example the title of this PR should have been `Logical operators`.

Should I create a new PR?

bugron commented:
@vahmelk99 nope, just have that noted for future PRs. Thanks.

vahmelk99 commented:
Please review my translation one more time. Thank you for your time. Hope I don't miss anything) /done

bugron commented:
@vahmelk99 it seems that you need to sign the CLA. That's a routine for this project.

vahmelk99 commented:
/done

bugron commented:
Did you sign the CLA? Because GitHub still shows you didn't.
<img width="842" alt="image" src="https://user-images.githubusercontent.com/13225220/129204576-8cb3909b-99c5-4c71-869d-22b2e8254c58.png">

bugron commented on 1-js/02-first-steps/11-logical-operators/3-alert-1-null-2/solution.md:
````diff
@@ -1,6 +1,6 @@
 
 ```js run
-alert( 1 && null && 2 );
+alert( 1 && null && 2 ); 
````
```suggestion
alert( 1 && null && 2 );
```

bugron commented on 1-js/02-first-steps/11-logical-operators/4-alert-and/task.md:
````diff
@@ -2,11 +2,10 @@ importance: 3
 
 ```js
-alert( alert(1) && alert(2) );
+alert( alert(1) && alert(2) ); 
````
```suggestion
alert( alert(1) && alert(2) );
```

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
````diff
@@ -1,24 +1,24 @@
 
 ```js
-result = a || b;
+result = a || b
````
```suggestion
result = a || b;
```

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -153,137 +153,135 @@ alert( true && false );  // false
 
 if (hour == 12 && minute == 30) {
-  alert( 'The time is 12:30' );
+  alert('Ժամը 12։30֊ն է')
```
```suggestion
  alert('Ժամը 12։30֊ն է');
```
Only translate strings please and don't do any stylistic changes.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -153,137 +153,135 @@ alert( true && false );  // false
-let hour = 12;
-let minute = 30;
+let hour = 12
+let minute = 30
```
```suggestion
let hour = 12;
let minute = 30;
```

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -153,137 +153,135 @@ alert( true && false );  // false
-if (1 && 0) { // evaluated as true && false
-  alert( "won't work, because the result is falsy" );
+if (1 && 0) { // հաշվվում է ինչպես true && false
+  alert('չի երևա, որովհետև արժեքը սխալական է')
```
```suggestion
  alert('չի երևա, որովհետև արժեքը սխալական է');
```

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -153,137 +153,135 @@ alert( true && false );  // false
+// Եթե առաջին օպերանդը ճշմարիտ է,
+// ապա ԵՎ֊ը վերադարձնում է երկրոդ օպերանդը։
+alert(1 && 0) // 0
+alert(1 && 5) // 5
```
```suggestion
alert( 1 && 0 ); // 0
alert( 1 && 5 ); // 5
```

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
```diff
@@ -153,137 +153,135 @@ alert( true && false );  // false
+// Եթե առաջին օպերանդը սխալական է,
+// ապա ԵՎ֊ը վերադարձնում է այն։ Երկրորդ օպերանդը անտեսվում է։
+alert(null && 5) // null
+alert(0 && 'կապ չունի ինչ') // 0
```
```suggestion
alert( null && 5 ); // null
alert( 0 && "կապ չունի ինչ" ); // 0
```
Please make sure it is not your editor altering those examples. Also make sure to restore the original formatting of any example code in all files of this PR. Thank you.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
`````diff
@@ -153,137 +153,135 @@ alert( true && false );  // false
-````smart header="Precedence of AND `&&` is higher than OR `||`"
-The precedence of AND `&&` operator is higher than OR `||`.
+````smart header="ԵՎ `&&`֊ի նախապատվությունը (precedence) ավելի բարձր է քան ԿԱՄ `||`֊ինը"
+
`````
Let's get rid of this new line.

bugron commented on 1-js/02-first-steps/11-logical-operators/article.md:
````diff
@@ -153,137 +153,135 @@ alert( true && false );  // false
 }
 ```
 
+## ԵՎ "&&"֊ը գտնում է առաջին սխալական արժեքը
````
Please put this line in its original place.

bugron reviewed (changes requested):
@vahmelk99 please make sure there are no unnecessary line shifts here and there, also make sure to restore original formatting and new lines where mentioned. Thanks for you contribution.

vahmelk99 committed:
Update 1-js/02-first-steps/11-logical-operators/3-alert-1-null-2/solution.md

Co-authored-by: Arsen Melikyan <bugron@mail.ru>

vahmelk99 committed:
Update 1-js/02-first-steps/11-logical-operators/4-alert-and/task.md

Co-authored-by: Arsen Melikyan <bugron@mail.ru>

vahmelk99 committed:
Update 1-js/02-first-steps/11-logical-operators/article.md

Co-authored-by: Arsen Melikyan <bugron@mail.ru>

vahmelk99 committed:
Update 1-js/02-first-steps/11-logical-operators/article.md

Co-authored-by: Arsen Melikyan <bugron@mail.ru>

vahmelk99 committed:
Update 1-js/02-first-steps/11-logical-operators/article.md

Co-authored-by: Arsen Melikyan <bugron@mail.ru>

vahmelk99 committed:
Update 1-js/02-first-steps/11-logical-operators/article.md

Co-authored-by: Arsen Melikyan <bugron@mail.ru>

vahmelk99 committed:
Update 1-js/02-first-steps/11-logical-operators/article.md

Co-authored-by: Arsen Melikyan <bugron@mail.ru>

vahmelk99 committed:
Update 1-js/02-first-steps/11-logical-operators/article.md

Co-authored-by: Arsen Melikyan <bugron@mail.ru>

vahmelk99 committed:
article fixed

vahmelk99 commented:
@bugron thanks for your time.

vahmelk99 commented:
/done

bugron commented:
@vahmelk99 thanks for your efforts but unfortunately not all the comments are addressed. GitHub might hide some of them so please make sure everything is in order before we can move on to merging this PR. Thanks.

bugron reviewed (approved):
@vahmelk99 looks superb! Thanks again for your contribution! I've fixed a few line shifts and svg translation stuff and we're ready to merge this PR.

vahmelk99 commented:
> @vahmelk99 looks superb! Thanks again for your contribution! I've fixed a few line shifts and svg translation stuff and we're ready to merge this PR.

In the SVG file, I fixed the coordinates of the text, so there were not any overflowed texts.

bugron commented:
@vahmelk99 the thing is that SVG file should not be modified directly, there is a note about that in README.md. A special `images.yml` file is used to define SVG text translations and text positions. But its a bit tricky to see the results here because you'll also need to set up the server, run a gulp job to generate new images based on that yml file and start the project (local version of javascriptinfo.com) to check whether everything looks as intended or not. Cheers.
